src/gui/qgsfilewidget.cpp

```diff
@@ -359,16 +388,30 @@ void QgsFileDropEdit::setFilters( const QString &filters )
 
 QString QgsFileDropEdit::acceptableFilePath( QDropEvent *event ) const
 {
-  QString path;
+  QStringList paths;
   if ( event->mimeData()->hasUrls() )
   {
-    QFileInfo file( event->mimeData()->urls().first().toLocalFile() );
-    if ( ( mStorageMode == QgsFileWidget::GetFile && file.isFile() &&
-           ( mAcceptableExtensions.isEmpty() || mAcceptableExtensions.contains( file.suffix(), Qt::CaseInsensitive ) ) )
-         || ( mStorageMode == QgsFileWidget::GetDirectory && file.isDir() ) )
-      path = file.filePath();
+    for ( const auto url : event->mimeData()->urls() )
```

Comment:
Just a follow up - we can't use qAsConst since it's qt 5.7 and above only. I think we need to use std::as_const, and add a reference implementation of this for C++ <17 compilers (see http://en.cppreference.com/w/cpp/utility/as_const )